File: Code/SimpleBudget.API/Models/Payments/PaymentEditItemModel.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleBudget.API.Models
{
    public class PaymentEditItemModel
    {
        [Required]
        public string PaymentType { get; set; } = default!;

        [Required]
        public string Date { get; set; } = default!;

        public string? Company { get; set; }

        public string? Category { get; set; }

        [Required]
        public string Wallet { get; set; } = default!;

        public string? Description { get; set; }

        [Required]
        public decimal Value { get; set; }

        public bool Taxable { get; set; }

        public int? TaxYear { get; set; }

        public string? WalletTo { get; set; }

        public decimal? ValueTo { get; set; }

        public string? Person { get; set; }
    }
}

File: Code/client/src/api/models/PaymentEditItemModel.ts
export interface PaymentEditItemModel {
    paymentType: string;
    date: string;
    company: string | undefined | null;
    category: string | undefined | null;
    wallet: string;
    description: string | undefined | null;
    value: number;
    taxable: boolean;
    taxYear: number | undefined | null;
    walletTo: string | undefined | null;
    valueTo: number | undefined | null;
    person: string | undefined | null;
}